src/main/java/org/apache/commons/csv/CSVFormat.java

```diff
@@ -2280,6 +2279,27 @@ private void validate() throws IllegalArgumentException {
         }
     }
 
+    private Object readResolve() throws ObjectStreamException {
+        // check whether field duplicateHeaderMode (as of 1.10.0) was found in serialized form
+        // if not, set from boolean allowDuplicateHeaderNames
+        if (duplicateHeaderMode == null) {
+            setFieldValue("duplicateHeaderMode",
+                    allowDuplicateHeaderNames ? DuplicateHeaderMode.ALLOW_ALL : DuplicateHeaderMode.DISALLOW);
+        }
+        return this;
+    }
+
+    private void setFieldValue(String fieldName, Object value) {
+        try {
+            Field fld = getClass().getDeclaredField(fieldName);
+            fld.setAccessible(true); // make field non-final
+            fld.set(this, value);
+            fld.setAccessible(false);
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to set value of field " + fieldName, ex);

```

Comment:
RuntimeException is an anti-pattern, using IAE or ISE or another more descriptive exception.